consoles/VNC.pm

```diff
@@ -12,11 +12,13 @@ use Carp qw(confess cluck carp croak);
 use Data::Dumper 'Dumper';
 use Scalar::Util 'blessed';
 use OpenQA::Exceptions;
+use consoles::VMWare;
```

Comment:
Hm. I would prefer if we can avoid the VNC console to depend on the VMWare console. Can we use specific VMWare consoles that could inherit from VNC?

Reply:
I also thought about using inheritance. It would have been more work for nothing but if things generally work I can do that if it is required. Not sure whether it is the better design (if in doubt, I'd say https://en.wikipedia.org/wiki/Composition_over_inheritance).

Note that my initial thinking was that this is not a special console at all from the side of tests and the VNC console would figure out things on its own. And in my opinion it would still be nice if I could move code from https://github.com/Martchus/os-autoinst-distri-opensuse/commits/svirt-vmware-vnc-cfg/lib/susedistribution.pm to os-autoinst so all the VMWare-specific handling is at one place and this doesn't become a half test distribution specific feature (which is weirdly documented as backend variable here, despite the variable being used in the test distro). However, moving that code to os-autoinst means that from tests it shouldn't make a difference which VNC console to instantiate. So my initial idea of letting the normal VNC console handle it under the hood would actually still make sense. Note that `VMWare.pm` is conceptually not really its own console but just a helper for making the normal VNC console work with VMWare's VNC over WebSockets.

Reply:
Yes, I understood that. Well, ok, so that means that consoles::VNC is both the generic VNC console as well as the module combining different implementations I guess

Reply:
> so that means that consoles::VNC is both the generic VNC console

Not really because the generic VNC console is actually `consoles::vnc_base`. `consoles::VNC` is also just some module internally used by that console. Apparently our convention is to use lower case names for actual consoles and upper case names for "console helpers". Of course that's just a technicality. But that's my understanding of our conventions and that's also why I called the new module `consoles::VMWare`.